charts/vmclarity/templates/gateway/deployment.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "vmclarity.gateway.name" . }}
+  namespace: {{ .Release.Namespace | quote }}
+  labels: {{- include "vmclarity.gateway.labels.standard" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels: {{- include "vmclarity.gateway.labels.matchLabels" . | nindent 6 }}
+  replicas: {{ .Values.gateway.replicaCount }}
+  template:
+    metadata:
+      labels: {{- include "vmclarity.gateway.labels.standard" . | nindent 8 }}
+      annotations:
+        checksum/configmap: {{ include (print $.Template.BasePath "/gateway/configmap.yaml") . | sha256sum }}
+    spec:
+      {{- if .Values.gateway.affinity }}
+      affinity: {{- .Values.gateway.affinity | toYaml | nindent 8 }}
+      {{- end }}
+      {{- if .Values.gateway.nodeSelector }}
+      nodeSelector: {{- .Values.gateway.nodeSelector | toYaml | nindent 8 }}
+      {{- end }}
+      {{- if .Values.gateway.podSecurityContext.enabled }}
+      securityContext: {{- omit .Values.gateway.podSecurityContext "enabled" | toYaml | nindent 8 }}
+      {{- end }}
+      {{- if .Values.gateway.tolerations }}
+      tolerations: {{- .Values.gateway.tolerations | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: gateway
+          image: {{ include "vmclarity.images.image" ( dict "imageRoot" .Values.gateway.image "global" .Values.global ) }}
+          imagePullPolicy: {{ .Values.gateway.image.pullPolicy | quote }}
+          volumeMounts:
+            - name: gateway-config
+              mountPath: /etc/nginx/nginx.conf
+              subPath: nginx.conf
```

Comment:
why do we need the subPath here?

Reply:
Without the subPath the configmap would be mounted as a folder with files inside it. In this case I only want to override the nginx.conf file because the /etc/nginx directory has other things inside it.